File: src/main/java/com/kmatrokhin/uvbot/entities/LocationEntity.java
package com.kmatrokhin.uvbot.entities;

import com.kmatrokhin.uvbot.dto.Coordinates;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "location")
@Getter
@Setter
public class LocationEntity {
    @Id
    private UUID id = UUID.randomUUID();
    private Double latitude;
    private Double longitude;
    @ManyToOne
    @JoinColumn(name = "user_id")
    private UserEntity userEntity;
    private String name;
    private Float lastUvIndex;
    private Instant createdAt = Instant.now();

    public Coordinates coordinates() {
        return Coordinates.of(latitude, longitude);
    }
}

File: src/main/java/com/kmatrokhin/uvbot/entities/LocationEntity.kt
package com.kmatrokhin.uvbot.entities

import com.kmatrokhin.uvbot.dto.Coordinates
import jakarta.persistence.*
import java.time.Instant
import java.util.*

@Entity
@Table(name = "location")
class LocationEntity(
    @Id
    var id: UUID = UUID.randomUUID(),
    var latitude: Double,
    var longitude: Double,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var userEntity: UserEntity,
    var name: String? = null,
    var lastUvIndex: Float? = null,
    var createdAt: Instant = Instant.now()
) {
    fun coordinates(): Coordinates {
        return Coordinates(latitude, longitude)
    }
}
